src/Microsoft.AspNetCore.Mvc.Abstractions/Routing/AttributeRouteInfo.cs

```diff
@@ -26,5 +26,15 @@ public class AttributeRouteInfo
         /// route by provided route data.
         /// </summary>
         public string Name { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value that determines if this model participates in link generation.
+        /// </summary>
+        public bool SuppressForLinkGeneration { get; set; }
```

Comment:
I think we should drop the `For` from these names so `SuppressForLinkGeneration` and `SuppressPathMatching`